common/src/java/com/zimbra/common/service/ServiceException.java

```diff
@@ -347,7 +347,11 @@ public static ServiceException DEFEND_CALENDAR_RESOURCE_HARVEST(String calendarR
     }
 
     public static ServiceException AUTH_REQUIRED() {
-        return new ServiceException("no valid authtoken present", AUTH_REQUIRED, SENDERS_FAULT);
+        return AUTH_REQUIRED("no valid authtoken present", null);
+    }
+
+    public static ServiceException AUTH_REQUIRED(String message, Throwable cause) {
```

Comment:
I dislike this historical overloading of ServiceException instead of extending it.  This means that if we want to catch only exceptions related to auth problems, we have to catch ServiceException and introspect it instead of catching, say, AuthException.  I'd consider introducing an AuthRequiredException which extends ServiceException.